src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs

```diff
@@ -356,14 +357,66 @@ public static void AddTopLevelSuggestionsForCursorType(IntellisenseData.Intellis
                     }
                 }
 
+                FormulaValue[] parameters = callNode.Args.Children.Where(texlNode => texlNode.Kind != NodeKind.Error).Select(texlNode => texlNode switch
+                {
+                    StrLitNode strNode => FormulaValue.New(strNode.Value),
+                    NumLitNode numNode => FormulaValue.New(numNode.ActualNumValue),
+                    _ => null as FormulaValue
+                }).ToArray();
+
                 // If connector function has some suggestions, let's add them here
-                List<string> suggestions = info.Function.GetConnectorSuggestionsAsync(callNode, argPosition, CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
+                List<RecordValue> suggestions = info.Function.GetConnectorSuggestionsAsync(parameters, argPosition, CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
 
                 if (suggestions != null)
-                {
-                    foreach (string suggestion in suggestions)
+                {                    
+                    foreach (RecordValue suggestion in suggestions)
                     {
-                        AddSuggestion(intellisenseData, suggestion, SuggestionKind.Global, SuggestionIconKind.Other, DType.String, false);
+                        if (argPosition < info.Function.MaxArity - 1)
```

Comment:
>if (argPosition < info.Function.MaxArity - 1) [](http://example.com/codeflow?start=24&length=45)

What does this check mean? 
If we wrap the return result in a class, can it abstract this check as well?